File: scripts/day1-part1-simple.js
const fs = require("fs");

// Read input file
const input = fs
  .readFileSync("../inputs/day-1-input.txt", "utf-8")
  .trim()
  .split("\n");

function solvePart1(input) {
  let position = 50;
  let count = 0;

  for (const rotation of input) {
    const direction = rotation[0];
    const distance = Number(rotation.slice(1));

    if (direction === "L") {
      position -= distance;
    } else {
      position += distance;
    }

    // Handle wrap-around
    position = ((position % 100) + 100) % 100;

    // Check if at zero
    if (position === 0) {
      count++;
    }
  }

  return count;
}

// Benchmark
const iterations = 1;
const startTime = process.hrtime.bigint();

let result;
for (let i = 0; i < iterations; i++) {
  result = solvePart1(input);
}

const endTime = process.hrtime.bigint();
const totalNanos = Number(endTime - startTime);
const avgMicros = (totalNanos / iterations / 1000).toFixed(3);

console.log(`JavaScript (Simple Version)`);
console.log(`Answer: ${result}`);
console.log(`Average time: ${avgMicros} microseconds (${iterations} iterations)`);
console.log(`Total time: ${(totalNanos / 1000000).toFixed(2)} ms`);

File: scripts/day1-part1-simple.py
import time

# Read input file
with open("../inputs/day-1-input.txt") as f:
    rotations = f.read().strip().split("\n")

def solve_part1(rotations):
    position = 50
    count = 0

    for rotation in rotations:
        direction = rotation[0]
        distance = int(rotation[1:])
        
        if direction == "L":
            position -= distance
        else:
            position += distance
        
        # Handle wrap-around
        position = position % 100
        
        # Check if at zero
        if position == 0:
            count += 1
    
    return count

# Benchmark
iterations = 1
start_time = time.perf_counter()

result = 0
for _ in range(iterations):
    result = solve_part1(rotations)

end_time = time.perf_counter()
total_time = (end_time - start_time)
avg_micros = (total_time / iterations) * 1_000_000

print(f"Python (Simple Version)")
print(f"Answer: {result}")
print(f"Average time: {avg_micros:.3f} microseconds ({iterations} iterations)")
print(f"Total time: {total_time * 1000:.2f} ms")
